lib/src/configuration.dart

```diff
@@ -379,4 +383,69 @@ class SyncClientResetErrorHandler {
 }
 
 /// A client reset strategy where the user needs to fully take care of a client reset.
-typedef ManualSyncClientResetHandler = SyncClientResetErrorHandler;
+///
+/// If you set [ManualSyncClientResetHandler] callback as `syncClientResetErrorHandler` argument of [Configuration.flexibleSync],
+/// that will enable full control of moving unsynced changes to synced realm.
+/// {@category Sync}
+class ManualSyncClientResetHandler extends SyncClientResetErrorHandler {
```

Comment:
```suggestion
class ManualRecoveryHandler extends SyncClientResetErrorHandler {
```

Reply:
ManualClientResetExceptionHandler?

Reply:
Can the last person close the door? 😜 Seriously though, these are the names that we aligned on across other SDKs and we should not deviate from them.

Reply:
:) ok if that was aligned then we don't need to discuss it again